src/dataExplorer/context/tags.tsx

```diff
@@ -82,6 +91,53 @@ export const TagsProvider: FC<Prop> = ({children, scope}) => {
 
     setLoadingTagKeys(RemoteDataState.Loading)
 
+    if (isFlagEnabled('v2privateQueryUI') && isIOx) {
+      // user input is sanitized to avoid SQL injection
+      const sanitized = sanitizeSQLSearchTerm(searchTerm)
+      const queryTextSQL: string = `
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE
+            table_schema = 'iox'
+            AND data_type LIKE 'Dictionary%'
+            AND table_name = '${measurement}'
+            AND column_name ILIKE '%${sanitized}%'
+        LIMIT ${DEFAULT_LIMIT}
+      `
+      const newTags: Tags = {}
+      try {
+        const resp = await queryAPI(queryTextSQL, scope, {
+          language: LanguageType.SQL,
+          bucket,
+        } as QueryOptions)
+        const keys = (resp.parsed.table.columns?.column_name?.data ??
+          []) as string[]
```

Comment:
Some comment here as in other PRs re: whether the type casting is avoidable.

Reply:
Thanks for mentioning it!`resp.parsed.table.columns?.column_name?.data` has type in ` number[] | boolean[] | string[]`. The response data is tag keys and tag keys are strings, so casting the data to strings specifically here. For the tag values, it makes sense to not casting into `string[]` since tag values can be `number` or `boolean`